From 9156d7504c5ccd46fde2997dca207d3ffb374a39 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Wed, 25 Oct 2017 17:14:04 +0800
Subject: [PATCH] drm: bridge: dw-hdmi: fix ycbcr_420_allowed condition of
 RK3288

The version of RK3288 HDMI is 0x200a, it doesn't support ycbcr420 mode.

Change-Id: I65ee16ae79f42fd13514db8e5819db6d07db9d3d
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/gpu/drm/bridge/dw-hdmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/dw-hdmi.c b/drivers/gpu/drm/bridge/dw-hdmi.c
index 1cc9680d8059..5156271b0660 100644
--- a/drivers/gpu/drm/bridge/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/dw-hdmi.c
@@ -2989,7 +2989,7 @@ int dw_hdmi_bind(struct device *dev, struct device *master,
 	/* Unmask HPD, clear transitory interrupts, then unmute */
 	hdmi_writeb(hdmi, hdmi->phy_mask, HDMI_PHY_MASK0);
 
-	if (hdmi->dev_type == RK3288_HDMI && hdmi->version > 0x200a)
+	if (hdmi->dev_type == RK3288_HDMI && hdmi->version == 0x200a)
 		hdmi->connector.ycbcr_420_allowed = false;
 	else
 		hdmi->connector.ycbcr_420_allowed = true;
-- 
2.35.3

